kernel/vfs/src/path.rs

```diff
@@ -0,0 +1,109 @@
+/// A file system path.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct Path<'a> {
+    inner: &'a str,
```

Comment:
Allowing slashes in node names sounds like asking for trouble. Especially because the `std` fs API uses string paths and so we'd somehow have to escape slashes in node names.

Reply:
yeah we cannot allow slashes in fs node names, that's a fairly typical limitation.